R/textstat_simil.R

```diff
@@ -358,3 +379,37 @@ as.matrix.simil <- function(x, diag = 1.0, ...) {
     diag(df) <- diag
     df
 }
+
+#' Coerce a textstat_simil or textstat_dist to a sparse Matrix format
+#' 
+#' Coerce a \link{textstat_simil} or \link{textstat_dist} to a sparse \pkg{Matrix} format.
+#' @param x input textstat object
+#' @keywords textstat internal
+#' @export
+as.Matrix <- function(x) {
```

Comment:
The more **Matrix** package-like call would be to extend S4 methods for `as(x, "dgCMatrix")` and `as(x, "dsTMatrix")`.  This function is a bit indeterminate. My guess is that those wishing to coerce to these methods will be comfortable with the S4 functions extending **Matrix**.

Also note that this
```r
> textstat_simil(dfmat, method = "cosine", margin = "documents", tri = TRUE, min_simil = 0.50) %>%
+     as.Matrix()
5 x 5 sparse Matrix of class "dgCMatrix"
           2001-Bush 2005-Bush 2009-Obama 2013-Obama 2017-Trump
2001-Bush          1 0.5204355  0.5411649  0.5561972          .
2005-Bush          . 1.0000000  .          0.5163644          .
2009-Obama         . .          1.0000000  0.6373318          .
2013-Obama         . .          .          1.0000000          .
2017-Trump         . .          .          .                  1
```
runs afoul of the issues in https://github.com/quanteda/quanteda/issues/1455 for both the lower triangle and the thresholded out values, suggesting that these are zeroes when they are not.

A better approach would be an extension of `as(x, "dsTMatrix")` that avoids the intermediate dense matrix object here:
```r
> textstat_simil(dfmat, method = "cosine", margin = "documents", tri = FALSE) %>%
+     as.matrix() %>%
+     as("dsTMatrix")
5 x 5 sparse Matrix of class "dsTMatrix"
           2001-Bush 2005-Bush 2009-Obama 2013-Obama 2017-Trump
2001-Bush  1.0000000 0.5204355  0.5411649  0.5561972  0.4518935
2005-Bush  0.5204355 1.0000000  0.4575297  0.5163644  0.4349030
2009-Obama 0.5411649 0.4575297  1.0000000  0.6373318  0.4481950
2013-Obama 0.5561972 0.5163644  0.6373318  1.0000000  0.4546945
2017-Trump 0.4518935 0.4349030  0.4481950  0.4546945  1.0000000
```
and to disallow any sparse object method via **quanteda** at least that contains anything that looks like a zero. So we would only support the sparse symmetric version and we would have to guarantee that the diagonal is 1.0.

Reply:
Fixed in d94cbd7d.